### PR TITLE
Secure SSL defaults for HTTPC for OTP-25+

### PR DESCRIPTION
**Pull request by essen**
Fixes https://github.com/hexpm/hex_core/issues/62

Tests have not been ran at the time of PR open but I did run Erlang.mk tests using this code and they passed (although not many dep fetches happen via Hex in the test suite).

### CODE REVIEW

File: src/hex_http_httpc.erl [hunk 1]
```diff
@@ -20,24 +20,42 @@
...
-            io:format(
-                "[hex_http_httpc] using default ssl options which are insecure.~n"
-                "Configure your adapter with: "
-                "{hex_http_httpc, #{http_options => [{ssl, SslOpts}]}}~n"
-            );
+    HTTPOptions = if
+        HTTPS == true andalso SSLOpts0 == undefined ->
+            %% Add safe defaults if possible.
+            _ = code:load_file(public_key),
+            HasCacertsGet = erlang:function_exported(public_key, cacerts_get, 0),
```
[the remark this answers is not in the document]
Comment (essen) [reply src/hex_http_httpc.erl:36]: I will change it to a try/catch as that is simpler.

[the remark this answers is not in the document]
Comment (essen) [reply src/hex_http_httpc.erl]: Done.

**Comment by wojtekmach** [comment]
Thank you!

**Comment by essen** [comment]
Thank you! When the next release gets published I can merge my branch making Erlang.mk use hex_core also for fetching packages (instead of just for publishing).

**Comment by wojtekmach** [comment]
I wanted to do one more thing before next release which is something like this: https://github.com/beam-telemetry/telemetry/pull/134. I'm hoping to do that within a week, otherwise I'd cut new release with what we have which I think is just your patch. Thanks again, if you run any issues feel free to post here or create new issues, I'm on EEF slack and elixir/erlang forum too.

**Comment by essen** [comment]
No worries, I remembered that I can just target the current commit, so I did that and will pin to the next release whenever it comes out.

**Comment by wojtekmach** [comment]
hex_core v0.11.0 is out with this change, thank you again for the PR!

**Pull request merged**